react/week3/todos/src/components/todolist.js

```diff
@@ -0,0 +1,180 @@
+              <input type='text' ref={(value) => { this.input = value }} defaultValue={this.props.todo.description} required /> | {this.props.todo.deadline}
+              <input type="checkbox" checked={this.props.todo.checked} onChange={() => {this.props.handleCheckbox(this.props.index)}}/>
+              <button onClick={() => {this.props.deleteTodo(this.props.index)}}>Delete</button>
+              <button type='submit'>Update</button>
+            </form>
+          </FancyBorder>      
+        </li>      
+      )
+    }
+  
+    renderTodoList = () => {
+      return (
+        <li style={{textDecoration: this.props.todo.checked ? 'line-through' : 'none'}}>
+          <FancyBorder>
+            <label>{this.props.todo.description} | {this.props.todo.deadline}</label>
+            <input type="checkbox" checked={!!this.props.todo.checked} onChange={() => {this.props.handleCheckbox(this.props.index)}}/>
+            <button onClick={() => {this.props.deleteTodo(this.props.index)}}>Delete</button>
+            <button onClick={(event) => { 
+              event.stopPropagation();
+              this.toggleState()
+              }}>Edit</button>
+          </FancyBorder>       
+        </li> 
+      )
+    }
+    render() {
+      return (
+        <div>
+          { this.state.isEditing ? this.renderEditMode() : this.renderTodoList() }
+        </div>
+      )
+    }
+  }
+  
+  function fetchTodo() {
+    const URL = `https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw`
+    return fetch(URL).then(response => response.json())
+  }
+  
+  
+
+class TodoList extends React.Component {
+    state = {
+        description: '',
+        deadline: '',
+        checked: false,
+        todoList: []
+      }
+  
+    componentDidMount() {
+      this.fetchTodos()
+    }
+  
+    fetchTodos = () => {
+      fetchTodo().then(todo => {
+        this.setState({ todoList: todo})
+      })
+    }
+  
+    handleSubmit = (event) => {   
+      this.setState({
+        description: this.state.description,
+        deadline: this.state.deadline,
+        todoList: [...this.state.todoList, {description: this.state.description, deadline: this.state.deadline, checked:false} ]
+      })
+      event.preventDefault()
+    }
+  
+    handleDescriptionChange = (event) => {
+      this.setState({ description: event.target.value })
+    }
+  
+    handleDeadlineChange = (event) => {
+      let currentDate = new Date()
+      let selectedDate = new Date(event.target.value)
+      if(selectedDate < currentDate) {
+        alert('Your selected date is in the past')
+      }
+      this.setState({ deadline: event.target.value })
+    }
+  
+    deleteTodo = (index) => {
+      let todo = this.state.todoList.filter((element, elementIndex) => elementIndex !== index)
+      this.setState({ todoList: todo })
+    }
+    
+    editTodo = (index, event) => {
+      let todo = this.state.todoList[index]
+      todo['description'] = event
```

Comment:
I think this might not have been working as expected. To get the value of the input from the event, you need to access event.target.value
```suggestion
      todo['description'] = event.target.value
```

Reply:
Above where you use editTodo, if you modify your method to pass the event from the method argument instead of `this.input.value` I think you'll get the expected behaviour.